api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.catalog;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+
+public interface Catalog {
+  /**
+   * creates the table or throws {@link AlreadyExistsException}.
+   *
+   * @param schema the schema for this table, can not be null.
+   * @param spec the partition spec for this table, can not be null.
+   * @param tableProperties can be null or empty
+   * @param tableIdentifier an identifier to identify this table in a namespace.
+   * @return Table instance that was created
+   */
+  Table createTable(
```

Comment:
`dropTable` passes options after the identifier, but this is before the identifier. Lets standardize on the identifier first.

Reply:
Done.